pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -180,14 +180,8 @@ func (exp *flowExporter) initFlowExporter(collectorAddr string, collectorProto s
 		if err != nil {
 			return err
 		}
-		// Currently, supporting only IPv4 for Flow Aggregator.
-		ip := hostIPs[0].To4()
-		if ip != nil {
-			// Update the collector address with resolved IP of flow aggregator
-			collectorAddr = net.JoinHostPort(ip.String(), defaultIPFIXPort)
-		} else {
-			return fmt.Errorf("resolved Flow Aggregator address %v is not supported", hostIPs[0])
-		}
+		// Update the collector address with resolved IP of flow aggregator
+		collectorAddr = net.JoinHostPort(hostIPs[0].String(), defaultIPFIXPort)
```

Comment:
In the case of dual-stack, it seems we can pick either IPv4 or IPv6 addresses depending on the DNS LookUp output. What if we pick the IPv6 address and the node network is IPv4? In that scenario, the flow exporter will not be able to connect to the flow aggregator. I believe IPv4-in-IPv6 and IPv6-in-IPv4 is possible in dual stack. @lzhecheng any input here?

Reply:
Ideally, we should check the node network address format and pick the correct resolved address of the flow aggregator.

Reply:
Thanks, make sense to me, will try to check the node network address format to determine the v4/v6 address of flow aggregator.

Reply:
Yes, they are possible in a dual-stack setup.

Reply:
Checking both addresses and picking the "correct" one should be ok. However, I am not sure the case you describe is possible. I think in order to have dual-stack Services, the Nodes need to have dual-stack networking themselves.

BTW, it's still a bit weird to me that:
* we have a special case for `strings.Contains(collectorAddr, flowAggregatorDNSName)`. I thought we had removed that special case.
* the DNS lookup is not handled by the go-ipfix library